src/main/java/at/hannibal2/skyhanni/features/inventory/caketracker/CakeTracker.kt

```diff
@@ -0,0 +1,383 @@
+        }
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!isEnabled()) return
+        if (inAuctionHouse) {
+            unobtainedCakesDisplayed = getCakeTrackerData()?.let { data ->
+                InventoryUtils.getItemsInOpenChest().onEach { cakeItem ->
+                    cakeNamePattern.matchMatcher(cakeItem.stack.displayName) {
+                        group("year").toInt().takeIf {
+                            it !in data.cakesOwned
+                        }?.let {
+                            cakeItem highlight config.auctionHighlightColor.toChromaColor()
+                        }
+                    }
+                }.isNotEmpty()
+            } ?: false
+        }
+        if (inCakeInventory) checkInventoryCakes()
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        knownCakesInCurrentInventory.clear()
+        val inventoryName = event.inventoryName
+        if (cakeContainerPattern.matches(inventoryName)) {
+            if (cakeBagPattern.matches(inventoryName)) inCakeBag = true
+            knownCakesInCurrentInventory = event.inventoryItems.values.mapNotNull { item ->
+                cakeNamePattern.matchMatcher(item.displayName) {
+                    groupOrNull("year")?.formatInt()?.let {
+                        addCake(it)
+                        it
+                    }
+                }
+            }.toMutableList()
+            inCakeInventory = true
+            timeOpenedCakeInventory = SimpleTimeMark.now()
+            tracker.firstUpdate()
+        }
+        if (auctionBrowserPattern.matches(inventoryName)) {
+            inAuctionHouse = true
+            searchingForCakes = auctionCakeSearchPattern.matches(inventoryName)
+        } else inAuctionHouse = false
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        inCakeBag = false
+        inCakeInventory = false
+        knownCakesInCurrentInventory.clear()
+        inAuctionHouse = false
+        unobtainedCakesDisplayed = false
+        searchingForCakes = false
+    }
+
+    @SubscribeEvent
+    fun onSecondPassed(event: SecondPassedEvent) {
+        if (!isEnabled()) return
+        val sbTimeNow = SkyBlockTime.now()
+        if (currentYear == sbTimeNow.year) return
+        if (sbTimeNow.month == 12 && sbTimeNow.day >= 29) {
+            currentYear = sbTimeNow.year
+            recalculateMissingCakes()
+        } else currentYear = sbTimeNow.year - 1
+    }
+
+    private fun checkInventoryCakes() {
+        if (timeOpenedCakeInventory.passedSince() < 500.milliseconds) return
+        val currentYears = InventoryUtils.getItemsInOpenChest().mapNotNull { item ->
+            cakeNamePattern.matchMatcher(item.stack.displayName) {
+                group("year")?.toInt()
+                cakeNamePattern.matchGroup(item.stack.displayName, "year")?.toInt()
```

Comment:
i think the suggestion i did got applied wrong here, this line is supposed to replace the matchMatcher